js/lib/index.js

```diff
@@ -0,0 +1,703 @@
+  }
+
+  /**
+   * Fetch a Work from the global state, and return the value via a promise.
+   *
+   * The resulting object returned on the resolved promis will be structured
+   * like so:
+   *
+   * ```
+   * {
+   *   "title": "String", // a unique title of the work; the natural key
+   *   "ISWC": "International Standard Work Code", // may be null
+   *   "songwriter_publisher_splits": [
+   *     {
+   *       // all splits in this list must add up to 100
+   *       "split": <a positive int>,
+   *       "songwriter_publisher": {
+   *         "songwriter_name": "IndividualIdentity.name",
+   *         "publisher_name": "OrganizationalIdentity.name" // must be Type.PUBLISHER
+   *       }
+   *     },
+   *     // and more
+   *   ]
+   * }
+   * ```
+   *
+   * @param {string} title - the title of the work (its natural key)
+   *
+   * @returns {Promise} the promise for the work value
+   *
+   * @since 1.0.0
+   */
+  getWork (title) {
+    return _omiStateEntry(this._sawtoothRestUrl, Work, title)
+  }
+
+  getWorks () {
+    return _omiStateCursor(this._sawtoothRestUrl, Work)
+  }
+}
+
+/**
+ * A BatchStatus object provides a function to query for the current batch
+ * status of a submitted transaction. That is, whether or not the batch has been
+ * committed to the block chain.
+ */
+class BatchStatus {
+  /**
+   * Constructs a BatchStatus for a given batch id, with a url for querying the
+   * current status.
+   *
+   * @param {string} batchId - the batchId
+   * @param {string} statusUrl - the batch status url
+   */
+  constructor (batchId, statusUrl) {
+    /**
+     * @member {string}
+     */
+    this.batchId = batchId
+    this._statusUrl = statusUrl
+  }
+
+  /**
+   * Returns the batch status from a transaction submission.
+   *
+   * The status is returned via a promise.  If a timeout occurs, the promise is
+   * rejected.
+   *
+   * @param {number} [timeout=300000] - a timeout in millis, defaults to 5
+   * minutes
+   *
+   * @returns {Promise<string>} the status
+   */
+  get (timeout = 300000) {
+    return Promise.race([
+      _promiseGet(this._statusUrl).then((body) => body.data[this.batchId]),
+      new Promise((resolve, reject) => {
+        setTimeout(() => reject(new Error('Timeout occurred')), timeout)
+      })
+    ])
+  }
+}
+/**
+ * A cursor provides a way to traverse through the objects stored in global
+ * state.  Generally, this should not be created, but are returned from the
+ * various plural `get` methods.
+ *
+ * @see {@link OmiClient.getIndividuals}
+ * @see {@link OmiClient.getOrganizations}
+ * @see {@link OmiClient.getWorks}
+ * @see {@link OmiClient.getRecordings}
+ */
+class Cursor {
+  /**
+   * Constructs a cursor for a given sawtooth endpoint that supports the paging
+   * response semantic.
+   *
+   * @param {string} endpoint - the query endpoint
+   * @param {Cursor~xform} [xform] - an optional transform; defaults to
+   * identity
+   */
+  constructor (endpoint, xform = (x) => x) {
```

Comment:
I think, especially in this case, the clarity of that being a function is more obvious.